### PR TITLE
Feature/eodhp 219 add collection search to dynamic api

### PR DESCRIPTION
**Pull request by tjellicoe-tpzuk**
Added collection search extension as defined [here](https://github.com/stac-api-extensions/collection-search) to allow searching of collections by bbox and datetime. To test this extension the pull request making changes to the stac-fastapi repository also needs to be merged.

### CODE REVIEW

File: stac_fastapi/core/stac_fastapi/core/core.py [hunk 1]
```diff
@@ -944,3 +945,115 @@ async def get_queryables(
...
+        """
+
+        base_args = {
+            "bbox": bbox,
+            "limit": limit,
+        }
+
+        if datetime:
+            base_args["datetime"] = datetime
+        print(base_args)
```
Comment (sgillies-tpzuk) [inline stac_fastapi/core/stac_fastapi/core/core.py]: was this print for debug or is it intended for logging? Fine if so, but using the logging functionality is preferred. Probably not worth implementing just for this if not already configured.

File: stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py [hunk 2]
```diff
@@ -419,6 +501,58 @@ def apply_datetime_filter(search: Search, datetime_search):
...
+        """
+        if "eq" in datetime_search:
+            search = search.filter("range", **{"collection_start_time": {"lte": datetime_search['eq']}})
+            search = search.filter("range", **{"collection_end_time": {"gte": datetime_search['eq']}})            
+
+        else:
+            should = []
+            should.extend(
+                [
+                    Q("bool", filter=[
```
Comment (sgillies-tpzuk) [inline stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py]: Formatting looks weird here, have you run black over this?

**Review by sgillies-tpzuk — APPROVED** [review]

File: stac_fastapi/core/stac_fastapi/core/core.py (hunk 1 above)
Comment (tjellicoe-tpzuk) [reply stac_fastapi/core/stac_fastapi/core/core.py]: Good spot. No longer needed so I have removed it.

File: stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py (hunk 2 above)
Comment (tjellicoe-tpzuk) [reply stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py]: I hadn't but have now. I think it looks a bit better now.

**Pull request merged**